### PR TITLE
Coupling details

### PR DESCRIPTION
**Pull request by simsimkic**

### CODE REVIEW

File: CodeModel/CaDETModel/CodeItems/CaDETClass.cs [hunk 1]
```diff
@@ -42,34 +42,49 @@ public CaDETClass(string name, string fullName, string sourceCode): this(name)
 
         public List<CaDETClass> GetFieldLinkedTypes()
         {
-            return Fields.SelectMany(f => f.GetLinkedTypes()).ToList();
+            var instances = Fields.SelectMany(f => f.GetLinkedTypes()).ToList();
```
Comment (Luburic) [inline CodeModel/CaDETModel/CodeItems/CaDETClass.cs:45]: "instances" is not a good word, as it is part of the DE domain and ubiquitous language, and not the code model.

File: DataSetExplorer/Core/DataSets/CaDETToDataSetProjectBuilder.cs [hunk 2]
```diff
@@ -151,44 +150,67 @@ private static List<CoupledClassStrength> CountCouplingStrength(List<CaDETClass>
...
+            else
+            {
+                var couplingTypeAndSt = new Dictionary<CouplingType, int>();
+                couplingTypeAndSt.Add(cc.CouplingType, cc.CouplingStrength);
+                relatedInstances.Add(new RelatedInstance(cc.CoupledClass.FullName,
+                    GetCodeUrl(cc.CoupledClass.FullName), relationType, couplingTypeAndSt));
+            }
         }
 
         private List<RelatedInstance> FindInstancesThatReference(CaDETClass referencedClass)
```
Comment (Luburic) [inline DataSetExplorer/Core/DataSets/CaDETToDataSetProjectBuilder.cs:206]: This class is slowly growing into a god. Should these coupling items be part of the code model? You don't need to resolve this now, but maybe create an issue.

**Review by Luburic — CHANGES_REQUESTED** [review]
A few comments to examine inline.

File: DataSetExplorer/Core/DataSets/CaDETToDataSetProjectBuilder.cs (hunk 2 above)
Comment (simsimkic) [reply DataSetExplorer/Core/DataSets/CaDETToDataSetProjectBuilder.cs:206]: https://github.com/Clean-CaDET/platform/issues/99

File: CodeModel/CaDETModel/CodeItems/CaDETClass.cs [hunk 3]
```diff
@@ -42,34 +42,49 @@ public CaDETClass(string name, string fullName, string sourceCode): this(name)
...
+            var instances = Members.Where(m => m.Type is not CaDETMemberType.Constructor)
                 .SelectMany(m => m.GetLinkedReturnTypes()).ToList();
+            RemoveThisClassFromList(instances);
+            return instances;
         }
 
         public List<CaDETClass> GetMethodLinkedVariableTypes()
         {
-            return Members.Where(m=> m.Type is not CaDETMemberType.Property)
+            var instances = Members.Where(m => m.Type is not CaDETMemberType.Property)
```
Comment (Luburic) [inline CodeModel/CaDETModel/CodeItems/CaDETClass.cs:60]: Still using instances.

**Review by Luburic — APPROVED** [review]
One minor rename to perform and you can merge.

**Pull request merged**